syz-fuzzer/proc.go

```diff
@@ -247,16 +267,23 @@ func (proc *Proc) executeHintSeed(p *prog.Prog, call int) {
 	})
 }
 
-func (proc *Proc) execute(execOpts *ipc.ExecOpts, p *prog.Prog, flags ProgTypes, stat Stat) *ipc.ProgInfo {
-	info := proc.executeRaw(execOpts, p, stat)
-	calls, extra := proc.fuzzer.checkNewSignal(p, info)
+func (proc *Proc) execute(execOpts *ipc.ExecOpts, p *prog.Prog, flags ProgTypes,
+	stat Stat) (*ipc.ProgInfo, learning.ExecResult) {
+	ret := learning.ExecResult{
+		Coverage: 0,
+		TimeExec: 0,
+	}
+	info, time := proc.executeRaw(execOpts, p, stat)
+	ret.TimeExec = time
+	calls, extra, cov := proc.fuzzer.checkNewSignal(p, info)
+	ret.Coverage = cov
```

Comment:
Or better move ret initialization to here and skip the intermediate hops:
```
ret := learning.ExecResult{
    Coverage: cov,
    ExecTime: time,
}
```

Reply:
Done,